features/kinderen.feature

```diff
@@ -15,61 +16,188 @@ Functionaliteit: Kinderen van een persoon raadplegen
 
 
   Rule: Een kind wordt alleen teruggegeven als minimaal één gegeven in de identificatienummers (groep 01), naam (groep 02) of geboorte (groep 03) van het kind een waarde heeft.
-    # wanneer in een categorie kind alleen gegevens zijn opgenomen in groep 81 of 82, 85 en 86, wordt dit kind niet opgenomen in het antwoord
-    # wanneer een gegeven een onbekendwaarde heeft, zoals . (punt) bij geslachtsnaam en 00000000 bij geboortedatum, wordt die in deze regel beschouwd als leeg (geen waarde).
+    - Wanneer een gegeven een standaardwaarde heeft, zoals "." (punt) bij geslachtsnaam of "00000000" bij geboortedatum, geldt dat hier als het bestaan van een waarde en wordt het kind wel geleverd
+    - de gebruikte fields parameter in het request heeft geen invloed op de toepassing van deze regel, zolang er maar ten minste één gegeven van een kind wordt gevraagd
 
-    Scenario: kind volledig onbekend
+    # N.B. wanneer in een categorie kind alleen gegevens zijn opgenomen in groep 81 of 82, 85 en 86, wordt dit kind niet opgenomen in het antwoord
+    
+
+    @gba
+    Abstract Scenario: kind volledig onbekend
       Gegeven de persoon met burgerservicenummer 555550001 heeft de volgende kinderen in de registratie
-        | Categorie | Voornamen (02.10)  | Voorvoegsel (02.30) | Geslachtsnaam (02.40) | Geboortedatum (03.10) |
-        | 9         |                    |                     | .                     |                       |
-      En er zijn geen andere gegevens geregistreerd over de identificatienummers (groep 01), naam (groep 02) of geboorte (groep 03) in categorie 9
-      Als de persoon met burgerservicenummer 555550001 wordt geraadpleegd met fields=kinderen
-      Dan bevat het antwoord GEEN kinderen
+      | Categorie | Burgerservicenummer (01.20) | Voornamen (02.10) | Voorvoegsel (02.30) | Geslachtsnaam (02.40) | Geboortedatum (03.10) | Gemeente document (82.10) | Datum document (82.20) | Beschrijving document (82.30) | Ingangsdatum geldigheid (85.10) | Datum van opneming (86.10) |
+      | 9         |                             |                   |                     | <geslachtsnaam>       | <geboortedatum>       | 1926                      | 20040105               | D27894-2004-A782              | 20031107                        | 20040112                   |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                          |
+      | type                | RaadpleegMetBurgerservicenummer |
+      | burgerservicenummer | 999996150                       |
+      | fields              | kinderen.naam                   |
+      Dan heeft de persoon met burgerservicenummer '555550001' een kind met de volgende 'naam' gegevens
+      | naam          | waarde          |
+      | geslachtsnaam | <geslachtsnaam> |
+
+      Voorbeelden:
+      | geslachtsnaam | geboortedatum |
+      | .             |               |
+      |               | 00000000      |
```

Comment:
Begrijp ik hieruit goed dat er dan wel een kind geleverd wordt warvan de geslachtsnaam property leeg is ?

Reply:
Gelukkig, je hebt het goed begrepen. De RvIG stuurt GbaPersoon met de gegevens zoals ze in GBA-v staan, e de proxy maakt daar dan een OnbekendKind van.
Dit testgeval met een lege geslachtsnaam maar gevulde geboorte datum is trouwens alleen technisch mogelijk, hoort niet voor te komen.